.github/bin/error-log-analyzer.py

```diff
@@ -0,0 +1,519 @@
+causing additional errors.
+- `macro-call-in-module-params`: Errors that occured because of a macro call
+in a module parameter list. Those usually also contain the line delimiters
+of which the parser will be unaware causing syntax errors.
+ - `caused-by-macro-call-in-module-params`: Syntax errors that occur after a
+missing define. Analogous to `related-to-slang-validated-error`,
+those are syntax errors caused by earlier "missing" tokens.
+ - `unresolved-macro`: Errors that occured because of an unresolved macro call
+ - `test-designed-to-fail`: Errors that are intentional, present in e.g. ivtest
+ - `misc-preprocessor`: Errors that are not related to the above categories
+but are related to preprocessor keywords
+ - `misc-preprocessor-related`: Other errors that may have appeared because
+there was a preprocessor problem earlier.
+eg. a syntax error after an unresolved macro
+ - `standalone-header`: Errors that occured while parsing a header file, that
+really should not be parsed outside of its context where
+it is included
+ - `hit-preprocessor-failsafe`: During parsing a preprocessor-based
+`ifdef/elseif/else` decision tree a branch was selected that fails on
+purpose (present in rsd)
+ - `likely-unhandled-macro-call`: Above the syntax error there was a macro call
+that likely contained additional needed tokens that are not present now
+and the syntax is technically invalid.
+- `related-to-likely-unhandled-macro-call`: Syntax errors that occur after an
+unhandled macro was present near a syntax error
+"""
+
+
+SLANG_DEBUG_OUT = False
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument("path")
+parser.add_argument(
+        "--verible-path",
+        type=str,
+        required=False,
+        help="Verible project source root path"
+)
+args = parser.parse_args()
+root = args.path
+
+
+class State(Enum):
+    NORMAL = 0,
+    MODULE_DEFINE = 1,
+    SLANG_VERIFIED = 2,
+    MISC_PREPROCESSOR = 3,
+    MACRO_CALL_SYNTAX = 4
+
+
+# structure that contains the information about a particular error
+class ErrorContainer:
+    def __init__(
+            self,
+            project,
+            source_path,
+            line_number,
+            start_char,
+            end_char,
+            error_text,
+            category='undefined'
+    ):
+        self.project = project
+        self.source_path = source_path
+        self.line_number = line_number
+        self.start_char = start_char
+        self.end_char = end_char
+        self.category = category
+        self.error = error_text
+        self.slang_output = None
+
+    def __str__(self):
+        return f"Error in project: {self.project}\
+ of category: {self.category}\n \
+ source file: {self.source_path}\n \
+ on line: {self.line_number}, col: {self.start_char}-{self.end_char}\n \
+ full text:{self.error}"
+
+
+# error_dirs holds all the *-nonzeros directories in the provided root
+error_dirs = glob.glob(root+'/*-nonzeros')
+project_urls = sorted([
```

Comment:
TODO(future): have only one place where we store these, so that they are not in the shell script and here.